lib/tapioca/compilers/dsl/identity_cache.rb

```diff
@@ -64,6 +65,7 @@ module Dsl
       # ~~~
       class IdentityCache < Base
         extend T::Sig
+        include ColumnTypeHelper
```

Comment:
Can we use composition instead of inheritance here? Seen as we are extracting this out to some reusable code, we might want some other DSL generators to use the same logic without necessarily including a module. So it would be great if this was something like:
```ruby
@column_type_helper = ActiveRecordColumnTypeHelper.new(constant)
#...
@column_type_helper.type_for(column_name)
```
and if it was exported from the gem along with the DSL generation `Base` class.